airflow/providers/microsoft/azure/operators/data_factory.py

```diff
@@ -0,0 +1,149 @@
+        resource_group_name: Optional[str] = None,
+        factory_name: Optional[str] = None,
+        reference_pipeline_run_id: Optional[str] = None,
+        is_recovery: Optional[bool] = None,
+        start_activity_name: Optional[str] = None,
+        start_from_failure: Optional[bool] = None,
+        parameters: Optional[Dict[str, Any]] = None,
+        timeout: Optional[int] = 60 * 60 * 24 * 7,
+        check_interval: Optional[int] = 60,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.azure_data_factory_conn_id = azure_data_factory_conn_id
+        self.pipeline_name = pipeline_name
+        self.wait_for_completion = wait_for_completion
+        self.resource_group_name = resource_group_name
+        self.factory_name = factory_name
+        self.reference_pipeline_run_id = reference_pipeline_run_id
+        self.is_recovery = is_recovery
+        self.start_activity_name = start_activity_name
+        self.start_from_failure = start_from_failure
+        self.parameters = parameters
+        self.timeout = timeout
+        self.check_interval = check_interval
+
+    def execute(self, context: Dict) -> None:
+        self.hook = AzureDataFactoryHook(azure_data_factory_conn_id=self.azure_data_factory_conn_id)
+        self.log.info(f"Executing the '{self.pipeline_name}' pipeline.")
+        response = self.hook.run_pipeline(
+            pipeline_name=self.pipeline_name,
+            resource_group_name=self.resource_group_name,
+            factory_name=self.factory_name,
+            reference_pipeline_run_id=self.reference_pipeline_run_id,
+            is_recovery=self.is_recovery,
+            start_activity_name=self.start_activity_name,
+            start_from_failure=self.start_from_failure,
+            parameters=self.parameters,
+        )
+        self.run_id = vars(response)["run_id"]
+        # Push the ``run_id`` value to XCom regardless of what happens during execution. This allows for
+        # retrieval the executed pipeline's ``run_id`` for downstream tasks especially if performing an
+        # asynchronous wait.
+        context["ti"].xcom_push(key="run_id", value=self.run_id)
+
+        if self.wait_for_completion:
+            self.log.info(f"Waiting for pipeline run {self.run_id} to complete.")
+            self.hook.check_pipeline_run_status(
+                run_id=self.run_id,
+                check_interval=self.check_interval,
+                timeout=self.timeout,
+                resource_group_name=self.resource_group_name,
+                factory_name=self.factory_name,
+            )
+            self.log.info(f"Pipeline run {self.run_id} has completed successfully.")
+
+    def on_kill(self) -> None:
+        if self.run_id:
+            self.hook.cancel_pipeline_run(
+                run_id=self.run_id,
+                resource_group_name=self.resource_group_name,
+                factory_name=self.factory_name,
+            )
```

Comment:
is there a way we can indicate if the cancel pipeline was successful?

Reply:
I believe so. I imagine we could use the new functions here as well to make sure the pipeline is actually canceled. I'll implement and test.

Reply:
This check can be done via the new `wait_for_pipeline_run_status` function.